### PR TITLE
Create documentation and example setups for remote execution

### PR DESCRIPTION
**Pull request by aaronmondal**
These examples should help users test their infrastructure and worker images against different remote execution scenarios.

- - -
This change is [Reviewable](https://reviewable.io/reviews/TraceMachina/nativelink/1498)

### CODE REVIEW

**Review by aaronmondal — COMMENTED** [review]
Human-readable form at: https://nativelink-dkpg0cp5nz94.deno.dev/docs/rbe/remote-execution-examples

The directory with the examples is here: https://github.com/aaronmondal/nativelink/tree/toolchain-examples/toolchain-examples

+@allada +@MarcusSorealheis +@SchahinRohani @kubevalet

> *[Reviewable](https://reviewable.io/reviews/TraceMachina/nativelink/1498#-:-OCek9I2-zryPKTfDDcB:b17oz34)* status: 0 of 3 LGTMs obtained, and 0 of 19 files reviewed, and pending CI: Analyze (javascript-typescript), Analyze (python), Publish image, Publish nativelink-worker-init, Publish nativelink-worker-lre-cc, Web Platform Deployment / macos-14, Web Platform Deployment / ubuntu-24.04, asan / ubuntu-22.04, macos-13, pre-commit-checks, ubuntu-20.04 / stable, ubuntu-22.04, ubuntu-22.04 / stable, vale, windows-2022 / stable (waiting on @allada, @MarcusSorealheis, and @SchahinRohani)

**Review by allada — CHANGES_REQUESTED** [review]
> Reviewed 14 of 19 files at r1.
> *[Reviewable](https://reviewable.io/reviews/TraceMachina/nativelink/1498)* status: 0 of 3 LGTMs obtained, and 14 of 19 files reviewed, and pending CI: Bazel Dev / macos-13, Bazel Dev / macos-14, Bazel Dev / ubuntu-24.04, Cargo Dev / macos-13, Cargo Dev / ubuntu-22.04, Coverage, Installation / macos-13, Installation / macos-14, Installation / ubuntu-22.04, Local / ubuntu-22.04, NativeLink.com Cloud / Remote Cache / macos-14, NativeLink.com Cloud / Remote Cache / ubuntu-24.04, Publish image, Publish nativelink-worker-init, Publish nativelink-worker-lre-cc, Remote / large-ubuntu-22.04, Web Platform Deployment / macos-14, Web Platform Deployment / ubuntu-24.04, asan / ubuntu-22.04, docker-compose-compiles-nativelink (22.04), integration-tests (22.04), macos-13, ubuntu-20.04 / stable, ubuntu-22.04, ubuntu-22.04 / stable, windows-2022 / stable, and 2 discussions need to be resolved (waiting on @MarcusSorealheis and @SchahinRohani)

___
*[`toolchain-examples/MODULE.bazel` line 62 at r1](https://reviewable.io/reviews/TraceMachina/nativelink/1498#-OCeli_k3-Uxh9Mw0OYa:-OCeli_k3-Uxh9Mw0OYb:b7bwi1m) ([raw file](https://github.com/TraceMachina/nativelink/blob/bad7313ac11f324c5b77166a711395efc689166c/toolchain-examples/MODULE.bazel#L62)):*
> ```Starlark
> # - Download size: LARGE    Around 1.5 Gb for the linux toolchain.
> #
> # To test this toolchain, use for use with the config flag `--config=zig-cc`.
> ```

Wrong --config
___
*[`toolchain-examples/nativelink-config.json` line 42 at r1](https://reviewable.io/reviews/TraceMachina/nativelink/1498#-OCemOlI6jQx4oyi5hf6:-OCemOlL-H2WPzRUJ3FL:bgee2aa) ([raw file](https://github.com/TraceMachina/nativelink/blob/bad7313ac11f324c5b77166a711395efc689166c/toolchain-examples/nativelink-config.json#L42)):*
> ```JSON
>       "simple": {
>         "supported_platform_properties": {
>           "cpu_count": "minimum",
> ```

Should we removed unused ones?

**Review by aaronmondal — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/TraceMachina/nativelink/1498)* status: 0 of 3 LGTMs obtained, and 13 of 19 files reviewed, and pending CI: pre-commit-checks, and 2 discussions need to be resolved (waiting on @allada, @MarcusSorealheis, and @SchahinRohani)

___
*[`toolchain-examples/MODULE.bazel` line 62 at r1](https://reviewable.io/reviews/TraceMachina/nativelink/1498#-OCeli_k3-Uxh9Mw0OYa:-OCeuSXP0AkXXMs4yj3E:b-896fix) ([raw file](https://github.com/TraceMachina/nativelink/blob/bad7313ac11f324c5b77166a711395efc689166c/toolchain-examples/MODULE.bazel#L62)):*
<details><summary><i>Previously, allada (Nathan (Blaise) Bruer) wrote…</i></summary><blockquote>

Wrong --config
</blockquote></details>

Done.
___
*[`toolchain-examples/nativelink-config.json` line 42 at r1](https://reviewable.io/reviews/TraceMachina/nativelink/1498#-OCemOlI6jQx4oyi5hf6:-OCeuAQe--9cDJjYVkdp:bqti7d) ([raw file](https://github.com/TraceMachina/nativelink/blob/bad7313ac11f324c5b77166a711395efc689166c/toolchain-examples/nativelink-config.json#L42)):*
<details><summary><i>Previously, allada (Nathan (Blaise) Bruer) wrote…</i></summary><blockquote>

Should we removed unused ones?
</blockquote></details>

Might be a good idea. Should we just remove all of them? It might be easier to test things if the scheduler doesn't match against anything.

**Comment by MarcusSorealheis** [comment]
I'm reviewing but just need one more day.

File: toolchain-examples/.bazelrc [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# Don't use the host's default PATH and LD_LIBRARY_PATH.
+build --incompatible_strict_action_env
+
+# Use rules_python's builtin script to emulate a bootstrap python.
+build --@rules_python//python/config_settings:bootstrap_impl=script
+
+# Toolchain to verify remote execution with zig-cc.
+#
+# DON'T USE THIS FOR REAL WORKLOADS IT TAKES LIKE 10 YEARS TO BUILD A HELLO
```
Comment (MarcusSorealheis) [inline toolchain-examples/.bazelrc]: I don't think this needs to be said. Maybe:

`NOT RECOMMENDED FOR PRODUCTION IN MOST CASES DUE TO SLOWNESS`

The reality is for some customers hermeticity is more important than anything else.

Comment (aaronmondal) [reply toolchain-examples/.bazelrc]: Done.

**Review by aaronmondal — COMMENTED** [review]
> `toolchain-examples/`, like `deployment-examples/` need to live in an examples directory because that is where people look when they land

Agreed. It's currently a bit tricky to do this is because we never really differentiated between installation and testing of an installation. This change is a first step towards this and hopefully we can expand on this in the future to e.g. structure our documentation into "Installation" and "Testing" sections where installations cover various deployment methods and the tests work generically over all installation method.

> *[Reviewable](https://reviewable.io/reviews/TraceMachina/nativelink/1498#-:-OCoXtQw7SXMJTjtSgJY:bk79wli)* status: 1 of 3 LGTMs obtained, and 12 of 19 files reviewed, and pending CI: Analyze (javascript-typescript), Analyze (python), Bazel Dev / macos-13, Bazel Dev / macos-14, Bazel Dev / ubuntu-24.04, Cargo Dev / macos-13, Cargo Dev / ubuntu-22.04, Coverage, Installation / macos-13, Installation / macos-14, Installation / ubuntu-22.04, Local / ubuntu-22.04, NativeLink.com Cloud / Remote Cache / macos-14, NativeLink.com Cloud / Remote Cache / ubuntu-24.04, Publish image, Publish nativelink-worker-init, Publish nativelink-worker-lre-cc, Remote / large-ubuntu-22.04, Web Platform Deployment / macos-14, Web Platform Deployment / ubuntu-24.04, asan / ubuntu-22.04, docker-compose-compiles-nativelink (22.04), integration-tests (22.04), macos-13, pre-commit-checks, ubuntu-20.04 / stable, ubuntu-22.04, ubuntu-22.04 / stable, vale, windows-2022 / stable, and 4 discussions need to be resolved (waiting on @allada and @SchahinRohani)

___
*[`toolchain-examples/nativelink-config.json` line 42 at r1](https://reviewable.io/reviews/TraceMachina/nativelink/1498#-OCemOlI6jQx4oyi5hf6:-OCoYqJz9Sjc9QUftcCY:b-896fix) ([raw file](https://github.com/TraceMachina/nativelink/blob/bad7313ac11f324c5b77166a711395efc689166c/toolchain-examples/nativelink-config.json#L42)):*
<details><summary><i>Previously, aaronmondal (Aaron Siddhartha Mondal) wrote…</i></summary><blockquote>

Might be a good idea. Should we just remove all of them? It might be easier to test things if the scheduler doesn't match against anything.
</blockquote></details>

Done.
___
*[`web/platform/src/content/docs/docs/rbe/remote-execution-examples.mdx` line 28 at r2](https://reviewable.io/reviews/TraceMachina/nativelink/1498#-OCiI-2_Ey0bic8I_f2r:-OCoYnajCG3YNis7jF_D:b-896fix) ([raw file](https://github.com/TraceMachina/nativelink/blob/7a3032e68aadad1185e48436860382c2d6735731/web/platform/src/content/docs/docs/rbe/remote-execution-examples.mdx#L28)):*
<details><summary><i>Previously, MarcusSorealheis (Marcus Eagan) wrote…</i></summary><blockquote>

This is fine for the MVP. I think we should add a TODO here to include two paths for users to choose their own adventures using either Nix or Bazelisk and a link to the installation/setup steps. Unfortunately for us, each Markdown needs to almost stand on its own at this stage because we are so new.
</blockquote></details>

Done.

**Review by aaronmondal — COMMENTED** [review]
> Reviewed 14 of 19 files at r1, 1 of 1 files at r2, 4 of 4 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/TraceMachina/nativelink/1498)* status: 1 of 3 LGTMs obtained, and all files reviewed, and pending CI: Analyze (javascript-typescript), Analyze (python), Bazel Dev / macos-13, Bazel Dev / macos-14, Bazel Dev / ubuntu-24.04, Cargo Dev / macos-13, Cargo Dev / ubuntu-22.04, Coverage, Installation / macos-13, Installation / macos-14, Installation / ubuntu-22.04, Local / ubuntu-22.04, NativeLink.com Cloud / Remote Cache / macos-14, NativeLink.com Cloud / Remote Cache / ubuntu-24.04, Publish image, Publish nativelink-worker-init, Publish nativelink-worker-lre-cc, Remote / large-ubuntu-22.04, Web Platform Deployment / macos-14, Web Platform Deployment / ubuntu-24.04, asan / ubuntu-22.04, docker-compose-compiles-nativelink (22.04), integration-tests (22.04), macos-13, pre-commit-checks, ubuntu-20.04 / stable, ubuntu-22.04, ubuntu-22.04 / stable, vale, windows-2022 / stable, and 4 discussions need to be resolved (waiting on @allada and @SchahinRohani)

**Review by aaronmondal — COMMENTED** [review]
> Dismissed @MarcusSorealheis and @allada from 4 discussions.
> *[Reviewable](https://reviewable.io/reviews/TraceMachina/nativelink/1498)* status: 1 of 3 LGTMs obtained, and all files reviewed, and pending CI: Analyze (javascript-typescript), Analyze (python), Bazel Dev / macos-13, Bazel Dev / macos-14, Bazel Dev / ubuntu-24.04, Cargo Dev / macos-13, Cargo Dev / ubuntu-22.04, Coverage, Installation / macos-13, Installation / macos-14, Installation / ubuntu-22.04, Local / ubuntu-22.04, NativeLink.com Cloud / Remote Cache / macos-14, NativeLink.com Cloud / Remote Cache / ubuntu-24.04, Publish image, Publish nativelink-worker-init, Publish nativelink-worker-lre-cc, Remote / large-ubuntu-22.04, Web Platform Deployment / macos-14, Web Platform Deployment / ubuntu-24.04, asan / ubuntu-22.04, docker-compose-compiles-nativelink (22.04), integration-tests (22.04), macos-13, pre-commit-checks, ubuntu-20.04 / stable, ubuntu-22.04, ubuntu-22.04 / stable, vale, windows-2022 / stable (waiting on @allada and @SchahinRohani)

**Review by SchahinRohani — CHANGES_REQUESTED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/TraceMachina/nativelink/1498)* status: 1 of 3 LGTMs obtained, and all files reviewed, and pending CI: Bazel Dev / macos-13, Cargo Dev / macos-13, Coverage, Installation / macos-13, Installation / macos-14, Installation / ubuntu-22.04, NativeLink.com Cloud / Remote Cache / macos-14, Remote / large-ubuntu-22.04, docker-compose-compiles-nativelink (22.04), windows-2022 / stable, and 1 discussions need to be resolved (waiting on @allada and @SchahinRohani)

___
*[`web/platform/starlight.conf.ts` line 88 at r3](https://reviewable.io/reviews/TraceMachina/nativelink/1498#-OCoaCD8BPYC54zE1XFC:-OCoaCD92k-qmEXFzPRH:bxucz18) ([raw file](https://github.com/TraceMachina/nativelink/blob/2a2f7293d41776be8449b75d755edbc47b13a050/web/platform/starlight.conf.ts#L88)):*
> ```XML
>     },
>     {
>       label: "Testing remote execution",
> ```

Let's capitalize the first letters, so it's in sync with the all other labels

**Review by SchahinRohani — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/TraceMachina/nativelink/1498#-:-OCobnBpENGs2pbMltkV:bnfp4nl)* status: 2 of 3 LGTMs obtained, and all files reviewed, and pending CI: Analyze (javascript-typescript), Publish image, Publish nativelink-worker-init, Web Platform Deployment / macos-14, Web Platform Deployment / ubuntu-24.04, asan / ubuntu-22.04, pre-commit-checks, ubuntu-20.04 / stable, ubuntu-22.04, vale (waiting on @allada)

**Comment by MarcusSorealheis** [comment]
Please wait on Blaise’s review in this case.

**Comment by SchahinRohani** [comment]
I went trough the docs for remote execution and tested all targets.
So far all targets are passing in WSL! Great Work!

**Review by aaronmondal — COMMENTED** [review]
@allada Is this fine to merge?

> *[Reviewable](https://reviewable.io/reviews/TraceMachina/nativelink/1498#-:-ODlTy3y9gLYHh82enSh:b-9zh5s6)* status: 2 of 3 LGTMs obtained, and all files reviewed (waiting on @allada)

**Review by allada — APPROVED** [review]
:lgtm:

> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/TraceMachina/nativelink/1498#-:-OE-zN_413sv-1T_CCpi:bnfp4nl)* status: <span>:shipit:</span> complete! 3 of 3 LGTMs obtained, and all files reviewed

**Pull request merged**